sermant-plugins/sermant-flowcontrol/flowcontrol-common/src/main/java/io/sermant/flowcontrol/common/handler/retry/policy/RetryOnSamePolicy.java

```diff
@@ -63,12 +69,14 @@ public boolean isRetry() {
     }
 
     @Override
-    public Object getLastRetryServer() {
-        return lastRetryInstance;
+    public List<Object> getAllRetryInstance() {
+        return retryInstance;
     }
 
     @Override
     public void update(Object instance) {
-        this.lastRetryInstance = instance;
+        if (CollectionUtils.isEmpty(retryInstance)) {
+            retryInstance.add(instance);
```

Comment:
Why is the instance transferred only when the list is empty?

Reply:
done